### DISCUSSION TITLE
ListHandler not firing

### DISCUSSION THREAD

**Discussion by osdan-madrid** [serenity-is/Serenity#6388] [Help] [answered]
Hello everyone, I hope someone can help me with something that is perhaps something simple.

These are the tables I use.

Kermess is the main table.
Kermess_Clients and Kermess_Items are child tables.
Kermess_Items_Details is a grandchild table

<img width="467" alt="image" src="https://user-images.githubusercontent.com/42551335/166942848-92aac9ac-8015-4a57-8890-e2cd47843264.png">

And I have created the screens in that sequence

Kermess
![image](https://user-images.githubusercontent.com/42551335/166943438-8a36d808-4eab-4b87-b1d7-effe10c203d5.png)

Kermess_Items
![image](https://user-images.githubusercontent.com/42551335/166943641-bc0f7563-b64d-4952-be2c-262cf97cd47c.png)

Kermess_Clients
![image](https://user-images.githubusercontent.com/42551335/166943784-31c52e52-1021-47ab-83c1-4c827d0e1c09.png)

Up to this point, I can save and display information perfectly

In the last view, which could be inside any of the child views.

![image](https://user-images.githubusercontent.com/42551335/166944125-6fc4b734-f509-4b4b-9f58-d3b73fc39c4d.png)
![image](https://user-images.githubusercontent.com/42551335/166944195-695f0e2b-f5f7-4986-a7bb-86f99846d309.png)

![image](https://user-images.githubusercontent.com/42551335/166944688-fa605b38-b39d-40c8-af01-638fb8a95a3c.png)

If I add any record from this last view, the information registers in the database correctly.
<img width="253" alt="image" src="https://user-images.githubusercontent.com/42551335/166944915-898093cd-f9d1-47f3-8a57-330276c5b90d.png">

But if I even quit and go into another module, and go back to the record I just added, the view doesn't load the information.
![image](https://user-images.githubusercontent.com/42551335/166944195-695f0e2b-f5f7-4986-a7bb-86f99846d309.png)

I tried overriding the ListHandler's OnBeforeExecuteQuery, but it doesn't stop at the breakpoint when it starts loading the views.
<img width="643" alt="image" src="https://user-images.githubusercontent.com/42551335/166945304-fd200810-b3b0-462b-91fe-621e9ce86d51.png">

Someone who could guide me? Thanks in advance

**Comment by wezmag** [comment]
if you use `MasterDetailRelation` for Kermess Item and Kermess Item Detail, the details was populated in the [MasterDetailRelationBehavior](https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Net.Services/RequestHandlers/IntegratedFeatures/MasterDetail/MasterDetailRelationBehavior.cs).
That's why the `KermessItemsDetailsListHandler` was not called.

**Comment by osdan-madrid** [comment]
Thanks for answering @wezmag 

Just like you mentioned, I'm using it

```
 [DisplayName("Asocia Productos-Participantes"), MasterDetailRelation(foreignKey: "KermessItemId"), NotMapped]
        public List<KermessItemsDetailsRow> ClientesProductosList
        {
            get => fields.ClientesProductosList[this];
            set => fields.ClientesProductosList[this] = value;
        }
```

Although I don't understand why, for the Kermess and Kermess_Item tables, there it enters the ListHandler event, there it shows the information.

Anyway, what should I change to make it work properly? Or could it be that the problem is because you are making nested relationships at that level?

This is my code, but it's not working to retrieve items.

KermessItemsForm

`        [KermessItemsDetailsEditor]
        public List<KermessItemsDetailsRow> ClientesProductosList { get; set; }`

KermessItemsDetailsEditor
```
namespace Internal.Kermess {

    @Serenity.Decorators.registerEditor('Internal.Kermess.KermessItemsDetailsEditor')
    export class KermessItemsDetailsEditor extends Serenity.Extensions.GridEditorBase<KermessItemsDetailsRow> {
        protected getColumnsKey() { return KermessItemsDetailsColumns.columnsKey; }
        protected getDialogType() { return KermessItemsDetailsDialog; }
        protected getLocalTextPrefix() { return KermessItemsDetailsRow.localTextPrefix; }

        constructor(container: JQuery) {
            super(container);
        }

    }
}
```

KermessItemsDetailsRow
```
        [DisplayName("Kermess Item"), NotNull, ForeignKey("[dbo].[Kermess_Items]", "Id"), LeftJoin("jKermessItem")]
        public Int32? KermessItemId
        {
            get => fields.KermessItemId[this];
            set => fields.KermessItemId[this] = value;
        }
```

Or maybe do I have to use other way to do that?

Thanks in advance

**Reply by stixoffire** [reply]
Add (MinSelectLevel.Always) to your name field and also your id field for the Foreign Table Items.
Also any additional fields you want to be present in the other form Add LookupInclude() .

**Comment by osdan-madrid** [comment, marked as answer]
Thanks for your answers, for your time.

I could fix this situation with [this](https://github.com/serenity-is/Serenity/discussions/5959).
